File: src/main/java/com/example/leaderboard/domain/score/repository/ScoreRepository.java
package com.example.leaderboard.domain.score.repository;

import com.example.leaderboard.domain.score.entity.PuzzleScore;
import org.springframework.data.jpa.repository.JpaRepository;

import java.time.LocalDateTime;

public interface ScoreRepository extends JpaRepository<PuzzleScore, Long> {
    long countByUserIdAndSubmittedAtAfter(Long userId, LocalDateTime after);
}

File: src/main/kotlin/com/example/leaderboard/domain/score/repository/ScoreRepository.kt
package com.example.leaderboard.domain.score.repository

import com.example.leaderboard.domain.score.entity.PuzzleScore
import org.springframework.data.jpa.repository.JpaRepository
import java.time.LocalDateTime

interface ScoreRepository : JpaRepository<PuzzleScore, Long> {
    fun countByUserIdAndSubmittedAtAfter(userId: Long, after: LocalDateTime): Long
}
